docs/products/worldcoin-state-replication/using-world-id-zero-knowledge-proofs-in-avalanche-dapp.mdx

```diff
@@ -0,0 +1,461 @@
+          ['uint256', 'string'],
+          [
+            hashToField(
+              utils.solidityPack(['string'], [config.WORLDCOIN_APP_ID]),
+            ).toHexString(),
+            'your-action',
+          ],
+        ),
+      )
+
+      const unpackedProof = utils.defaultAbiCoder.decode(
+        ['uint256[8]'],
+        zkProof.proof,
+      )[0]
+
+      const txBody = getProveIdentityTxBody(
+        zkProof.merkle_root,
+        hashToField(provider.address).toHexString(),
+        zkProof.nullifier_hash,
+        externalNullifierHash.toHexString(),
+        unpackedProof.map((el: unknown) => BigNumber.from(el).toHexString()),
+      )
+
+      const tx = await provider?.signAndSendTx?.({
+        to: config?.[
+          `QUERY_VERIFIER_CONTRACT_ADDRESS_${selectedChainToPublish}`
+        ],
+        ...txBody,
+      })
+
+      verificationSuccessTx.set((tx as EthTransactionResponse).transactionHash)
+
+      navigate(RoutesPaths.authSuccess)
+    } catch (error) {
+      ErrorHandler.process(error)
+    }
+
+    setIsSubmitting(false)
+  }, [
+    zkProof,
+    getProveIdentityTxBody,
+    navigate,
+    provider,
+    selectedChainToPublish,
+    verificationSuccessTx,
+  ])
+```
+
+Finally, after transaction has been added to the block, we will show its hash to the user using the following code:
+
+```ts title="src/pages/AuthSuccess/AuthSuccess.tsx"
+    <div className='auth-success'>
+      <div className='auth-success__header'>
+        <div className='auth-success__header-icon-wrp'>
+          <Icon className='auth-success__header-icon' name={ICON_NAMES.check} />
+        </div>
+        <h2 className='auth-success__header-title'>{`Proof Submitted`}</h2>
+      </div>
+
+      <div className='auth-success__card'>
+        <span className='auth-success__card-title'>{`Check transaction`}</span>
+
+        <a
+          className='auth-success__copy-field-wrp'
+          href={provider?.getTxUrl?.(
+            config.SUPPORTED_CHAINS_DETAILS[config.DEFAULT_CHAIN],
+            verificationSuccessTx.get,
+          )}
+          target='_blank'
+          rel='noreferrer'
+        >
+          <span className='auth-success__copy-field'>
+            {abbrCenter(verificationSuccessTx.get, 10)}
+            <Icon
+              className='auth-success__copy-field-icon'
+              name={ICON_NAMES.externalLink}
+            />
+          </span>
+        </a>
+      </div>
+        <AppButton
+        className='auth-success__return-btn'
+        routePath={'/'}
+        text={`RETURN HOME`}
+        size={`large`}
+      />
+    </div>
+```
+
+At this point user has passed the verification and you can be sure, that it's the real humant and not a bot and provide him the full access to your dApp.
+
+If you want to proceed further, we recommend you to clone the front-end from the [GitHub](https://github.com/rarimo/web-client-worldcoin-id/tree/main), make the changes you need and use it for your purposes.
```

Comment:
Instead of "proceed further" - "tweak or expand this example", also add a link to contracts